src/components/sidebar/SafeList/index.tsx

```diff
@@ -59,13 +64,55 @@ export const _shouldExpandSafeList = ({
 const MAX_EXPANDED_SAFES = 3
 const NO_SAFE_MESSAGE = 'Create a new safe or add'
 
+export const addActionsToTxs = (
+  chainId: string,
+  address: string,
+  page: TransactionListPage,
+  isSafeOwned: boolean,
+  txsActionsByChain: Record<string, Record<string, SafeActions>>,
+  walletAddress?: string,
+) => {
+  if (page.results.length === 0) return txsActionsByChain
```

Comment:
I'd suggest not mutating the object but returning a modified clone of it instead.